crates/matrix-sdk/src/sync.rs

```diff
@@ -17,6 +17,14 @@ impl Client {
         response: sync_events::v3::Response,
     ) -> Result<SyncResponse> {
         let response = self.base_client().receive_sync_response(response).await?;
+        self.handle_sync_response(response).await
+    }
+
+    #[tracing::instrument(skip(self, response))]
+    pub(crate) async fn handle_sync_response(
+        &self,
+        response: SyncResponse,
+    ) -> Result<SyncResponse> {
```

Comment:
It is needed because of the following in `sliding_sync.rs`, allowing both of them to use the same internal method after converating `v3::Response` and `v4::Response` respectively.

```
impl Client {
    /// ...
    #[tracing::instrument(skip(self, response))]
    pub(crate) async fn process_sliding_sync(
        &self,
        response: v4::Response,
    ) -> Result<SyncResponse> {
        let response = self.base_client().process_sliding_sync(response).await?;
        tracing::info!("done processing on base_client");
        self.handle_sync_response(response).await
    }
}
```